api/proto/teleport/externalcloudaudit/v1/externalcloudaudit_service.proto

```diff
@@ -0,0 +1,113 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.externalcloudaudit.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/externalcloudaudit/v1/externalcloudaudit.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/externalcloudaudit/v1;externalcloudauditv1";
+
+// ExternalCloudAuditService provides methods to manage external cloud audit.
+// It's used only in Teleport Cloud for feature called 'bring your own audit bucket'.
+//
+// Creating ExternalCloudAudit does not enable external cloud audit in cluster,
+// it's working as a "draft". Test connection can be executed on draft.
+// To use ExternalCloudAudit in cluster call method PromoteToClusterClusterExternalCloudAudit,
+// which will clone existing ExternalCloudAudit into new resource.
+// Promoted resource can't be mutated, it can be only disabled.
+// To change values in cluster ExternalCloudAudit, first to change "draft",
+// test connection and call PromoteToClusterClusterExternalCloudAudit again.
+//
+// Supporting ExternalCloudAudit as resources: "draft" and "cluster", allow us
+// test connection before actually enabling it in cluster.
+service ExternalCloudAuditService {
+  // GetDraftExternalCloudAudit returns the draft external cloud audit configuration resource.
+  rpc GetDraftExternalCloudAudit(GetDraftExternalCloudAuditRequest) returns (GetDraftExternalCloudAuditResponse);
+
+  // UpsertDraftExternalCloudAudit upserts a draft external cloud Audit resource.
+  rpc UpsertDraftExternalCloudAudit(UpsertDraftExternalCloudAuditRequest) returns (UpsertDraftExternalCloudAuditResponse);
```

Comment:
I'm guessing this method always sets the resource state to Draft.

And the Disable method does nothing to the ExternalCloudAudit resource, but does remove the ClusterExternalCloudAudit.

Is that it?

Reply:
yes, exactly